src/main/java/ir/sahab/cleanup/Cleanups.java

```diff
@@ -68,16 +65,25 @@ public void doAllQuietly() {
 
     public void doAll() throws IOException {
         boolean allSucceeded = true;
-        for (RunnableWithException<Exception> closeStatement : closeStatements) {
+        // Probably the first error was the cause of the problem for the rest of the operations and therefore
+        // the most important error to investigate the problem. For this reason, only the first error is
+        // returned after the entire clean-up operations.
```

Comment:
The comments you have provided in this method are not just implementation comments. They are important regarding to the method contract. So, remove these comments and instead describe them in Javadoc. Provide Javadoc for both doAll() and doAllQuitely() methods.